src/devices/MemoryLCD/LSxxxB7DHxx.cs

```diff
@@ -0,0 +1,256 @@
+
+            BytesPerLine = (PixelWidth + 7) / 8;
+
+            _lineNumberBuffer = Enumerable.Range(0, PixelHeight).Select(m => (byte)m).ToArray();
+            _frameBuffer = new byte[BytesPerLine * PixelHeight];
+
+            Init();
+        }
+
+        /// <summary>
+        /// Updates data of only one specified line.
+        /// </summary>
+        public void DataUpdate1Line(byte lineIndex, Span<byte> data, bool frameInversion = false)
+        {
+            if (data.Length != BytesPerLine)
+            {
+                throw new Exception($"Data writing period must be {PixelWidth}ck ({BytesPerLine} bytes)");
+            }
+
+            // m(1), ag(1), d(18), dummy(2)
+            Span<byte> outputBuffer = stackalloc byte[BytesPerLine + 4];
+
+            outputBuffer[0] = (byte)(ModeSelectionPeriodByte.Mode | (frameInversion ? ModeSelectionPeriodByte.FrameInversion : ModeSelectionPeriodByte.Dummy));
+            outputBuffer[1] = AgBytes.Table[lineIndex];
+            data.CopyTo(outputBuffer.Slice(2, BytesPerLine));
+
+            WriteSpi(outputBuffer);
+        }
+
+        /// <summary>
+        /// Updates arbitrary multiple lines data.
+        /// </summary>
+        public void DataUpdateMultipleLines(Span<byte> lineIndex, Span<byte> data, bool frameInversion = false)
+        {
+            if (data.Length != lineIndex.Length * BytesPerLine)
+            {
+                throw new Exception($"Data writing period must be {lineIndex.Length * PixelWidth}ck ({lineIndex.Length * BytesPerLine} bytes)");
+            }
+
+            // m(1), ag1(1), d1(18), dummy2(1), ag2(1), d2(18), ... , dummyn(1), agn(1), dn(18), dummy(2)
+            Span<byte> outputBuffer = stackalloc byte[2 + (2 + BytesPerLine) * lineIndex.Length];
+
+            outputBuffer[0] = (byte)(ModeSelectionPeriodByte.Mode | (frameInversion ? ModeSelectionPeriodByte.FrameInversion : ModeSelectionPeriodByte.Dummy));
+
+            for (int i = 0; i < lineIndex.Length; i++)
+            {
+                int from = BytesPerLine * i;
+                int to = 1 + (2 + BytesPerLine) * i;
+
+                outputBuffer[to] = AgBytes.Table[lineIndex[i] + 1];
+
+                data.Slice(from, BytesPerLine).CopyTo(outputBuffer.Slice(to + 1, BytesPerLine));
+            }
+
+            WriteSpi(outputBuffer);
+        }
+
+        /// <summary>
+        /// Maintains memory internal data (maintains current display).
+        /// </summary>
+        public void Display(bool frameInversion = false)
+        {
+            // m(1), dummy(1)
+            Span<byte> outputBuffer = stackalloc byte[2];
+
+            outputBuffer[0] = (byte)(frameInversion ? ModeSelectionPeriodByte.FrameInversion : ModeSelectionPeriodByte.Dummy);
+
+            WriteSpi(outputBuffer);
+        }
+
+        /// <summary>
+        /// Clears memory internal data and writes white.
+        /// </summary>
+        public void AllClear(bool frameInversion = false)
```

Comment:
Can you also add in the summary what does frameInversion mean or when to use it? Also it seems like this method could probably just be called Clear or ClearAll instead of AllClear?